python/activator/middleware_interface.py

```diff
@@ -1798,8 +1786,12 @@ def _filter_datasets(src_repo: Butler,
         known_datasets = set()
         for dataset_type in dataset_types:
             try:
-                subset = set(dest_repo.registry.queryDatasets(dataset_type, *args, **kwargs))
-            except lsst.daf.butler.registry.DataIdValueError as e:
+                # Okay to have empty results.
+                subset = set(dest_repo.query_datasets(dataset_type, explain=False, *args, **kwargs))
+            except MissingDatasetTypeError as e:
+                _log.debug("Pre-export query with args '%s' failed with %s", formatted_args, e)
+                # If dataset type never registered locally, then *any* such datasets are missing.
+            except DataIdValueError as e:
                 _log.debug("Pre-export query with args '%s' failed with %s", formatted_args, e)
                 # If dimensions are invalid, then *any* such datasets are missing.
```

Comment:
I don't know how I missed this before, but why not just merge the two branches:
```suggestion
            except (DataIdValueError, MissingDatasetTypeError) as e:
                _log.debug("Pre-export query with args '%s' failed with %s", formatted_args, e)
                # If dataset type or dimensions never registered locally, then *any* such datasets are missing.
```

Reply:
Agree that's better. Modified.